### PR TITLE
Clarification of the mutation and query interfaces and addition of the payload types and what they return as default.

### PR DESCRIPTION
**Pull request by aarona**
I promise not to mess this one up. If you want me to change something, do I just push another commit to master and those changes will show up here? I might need some hand holding with this so I can do it right.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,25 +133,38 @@ end
...
-1. userLogin
-1. userLogout
-1. userSignUp
-1. userUpdatePassword
-1. userSendResetPassword
+1. `userLogin(email: String!, password: String!): UserLoginPayload`
+1. `userLogout: UserLogoutPayload`
+1. `userSignUp(email: String!, password: String!, passwordConfirmation: String!, confirmSuccessUrl: String): UserSignUpPayload`
+
+   The parameter `confirmSuccessUrl` is optional unless you are using the `confirmable` plugin from Devise in your `resourse` class. If you have `confirmable` set up, you will have to provide it unless you have `config.default_confirm_success_url` set in `config/initializers/devise_token_auth.rb`.
```
Comment (aarona) [inline README.md]: I think I should keep model/class a standard here. Which would you prefer?

**Comment by mcelicalderon** [comment]
I think the original version is better. `current_resource` is already a term used to identify the resource making the request after it is authenticated  (e.g. `current_user`). Also careful with your git history, PR's should look much cleaner than this. Just make sure the branch master in your fork is up to date with ours, you can achieve this by having to separate remotes and doing a pull of ours into your local master and then create a separate branch for your change

**Comment by aarona** [comment]
Ok got it. Pull from your master into my local master and then push to my githubs master? I used `current_resource` because I was assuming that if you were to inherit from you would have access to that variable but you're saying that regardless of class type that `current_user` will be the `current_resource` even if the model name is `Admin` or `Employee` for example?

**Comment by aarona** [comment]
Also check the commit labeled `f21ee9a` as `5ed808c` is a quick update I made to this PR after I noticed some misspellings of mine afterwards.

**Comment by aarona** [comment]
It looks like I needed to update this branch with your master so I just did that. Did you look at the changes I made?

File: README.md [hunk 2]
```diff
@@ -156,25 +156,38 @@ The install generator can do this for you because it executes DTA installer.
...
+1. `userConfirmAccount(confirmationToken: String!, redirectUrl: String!): User`
+1. `userCheckPasswordToken(resetPasswordToken: String!, redirectUrl: String): User`
 
-The reason for having 2 queries is that these 2 are going to be accessed when clicking on
+The reason for having two queries is that these two are going to be accessed when clicking on
 the confirmation and reset password email urls. There is no limitation for making mutation
 requests using the `GET` method on the Rails side, but looks like there might be a limitation
 on the [Apollo Client](https://www.apollographql.com/docs/apollo-server/v1/requests/#get-requests).
 
+#### Types
```
Comment (mcelicalderon) [inline README.md]: I'm not sure we should document these types as they are just generated by the graphql gem. Listing the type names doesn't seem useful.

File: README.md [hunk 3]
```diff
@@ -156,25 +156,38 @@ The install generator can do this for you because it executes DTA installer.
...
+1. `UserLoginPayload`
+1. `UserLogoutPayload`
+1. `UserSendResetPasswordPayload`
+1. `UserSignUpPayload`
+1. `UserUpdatePasswordPayload`
+
+By default, each payload type returned from the mutations have a single field of type `User` aliased as `authenticable`.
```
Comment (mcelicalderon) [inline README.md]: This information is more useful and there might be a better way to explain the way we add default fields. Maybe in the section we explain how to add more fields or extend the default mutation. Also, we need to remain abstract about the model or type we use as it is not always `User`. Also `authenticable` is not an alias, as it is only the field name for any model.

File: README.md [hunk 4]
```diff
@@ -156,25 +156,38 @@ The install generator can do this for you because it executes DTA installer.
...
+1. `userSendResetPassword(email: String!, redirectUrl: String!): UserSendReserPasswordPayload`
 
 #### Queries
-1. userConfirmAccount
-1. userCheckPasswordToken
+1. `userConfirmAccount(confirmationToken: String!, redirectUrl: String!): User`
+1. `userCheckPasswordToken(resetPasswordToken: String!, redirectUrl: String): User`
 
-The reason for having 2 queries is that these 2 are going to be accessed when clicking on
+The reason for having two queries is that these two are going to be accessed when clicking on
```
Comment (mcelicalderon) [inline README.md]: 👍

File: README.md [hunk 5]
```diff
@@ -156,25 +156,38 @@ The install generator can do this for you because it executes DTA installer.
 See [Installation](#Installation) for details.
 
 ### Making Requests
-Here is a list of the available mutations and queries assuming your mounted model
+Here is a list of the available mutations queries, and types assuming your mounted model
```
Comment (mcelicalderon) [inline README.md]: Would also need to change if we don't document types

File: README.md [hunk 6]
```diff
@@ -156,25 +156,38 @@ The install generator can do this for you because it executes DTA installer.
...
-1. userLogin
-1. userLogout
-1. userSignUp
-1. userUpdatePassword
-1. userSendResetPassword
+1. `userLogin(email: String!, password: String!): UserLoginPayload`
+1. `userLogout: UserLogoutPayload`
+1. `userSignUp(email: String!, password: String!, passwordConfirmation: String!, confirmSuccessUrl: String): UserSignUpPayload`
+
+   The parameter `confirmSuccessUrl` is optional unless you are using the `confirmable` plugin from Devise in your `resource`'s model. If you have `confirmable` set up, you will have to provide it unless you have `config.default_confirm_success_url` set in `config/initializers/devise_token_auth.rb`.
```
Comment (mcelicalderon) [inline README.md]: Nice

**Comment by mcelicalderon** [comment]
Also, if you are looking for something to contribute to, I'd love to see a PR from you on some of the functionality. If you have the time, you could give it a go with the resend confirmation email mutation. Look at this as a reference https://github.com/lynndylanhurley/devise_token_auth/blob/c9bf116735d081ba8888d5098481d23882cd1e65/app/controllers/devise_token_auth/confirmations_controller.rb#L32

File: README.md (hunk 2 above)
Comment (aarona) [reply README.md]: Got it. I can remove that along with the mention of the types above where you commented them.

File: README.md (hunk 3 above)
Comment (aarona) [reply README.md]: Got it. For some reason, I thought the `UserType` was alised as `user:` (maybe it was in an older version?) but I see now its no longer doing this and is returned as `authenticable:`. I'll remove the types altogether.

**Comment by aarona** [comment]
I'd love to look into that feature addition today.

It looks like something got merged into master that's causing my PR to fail. I did a pull from `upstream` and had to merge it into my fork before I could push it which I did, now it looks like some tests are failing.

**Comment by mcelicalderon** [comment]
Not really, looks like DTA merged a breaking change into master an we test Rails and DTA edge. Will look into that later this week

**Comment by aarona** [comment]
Ok, it looks like this is passing now with those edge tests being optional. Also I have a `resend_confirmation` branch I'd like to do a pull request for.

**Pull request closed without merging**